src/components/dimension-item/dimension-item.tsx

```diff
@@ -0,0 +1,145 @@
+import { IconAdd16, IconSubtract16 } from '@dhis2/ui'
+import { useSortable } from '@dnd-kit/sortable'
+import { CSS } from '@dnd-kit/utilities'
+import React from 'react'
+import { DimensionItemBase } from './dimension-item-base'
+import { IconButton } from './icon-button'
+import type { SupportedDimensionType } from '@constants/dimension-types'
+
+interface DimensionItemProps {
+    id: string
+    draggableId?: string
+    name: string
+    dimensionType: SupportedDimensionType
+    valueType?: string
+    optionSet?: string | null
+    stageName?: string
+    disabled?: boolean
+    selected?: boolean
+}
+
+export const DimensionItem: React.FC<DimensionItemProps> = ({
+    id,
+    draggableId,
+    name,
+    dimensionType,
+    valueType,
+    optionSet,
+    stageName,
+    disabled,
+    selected,
+}) => {
+    // const dispatch = useDispatch()
```

Comment:
Note that when you start using things like `useDispatch` and `useSelector`, you want to be importing the type-aware `useAppSelector` and `useAppDispatch` instead. I believe I have some ESLint rules in place to guide you in the right direction for this too. See [here](https://redux-toolkit.js.org/tutorials/typescript#define-root-state-and-dispatch-types) and [here](https://redux-toolkit.js.org/tutorials/typescript#define-root-state-and-dispatch-types)

Reply:
Sounds good. I'll wait to do this as there is no point in adding any redux until the sidebar is implemented.